easybuild/easyblocks/b/bwise.py

```diff
@@ -0,0 +1,120 @@
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+@author: Ward Poelmans (Free University of Brussels)
+"""
+import glob
+import os
+import stat
+
+from easybuild.easyblocks.generic.makecp import MakeCp
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import adjust_permissions, apply_regex_substitutions, change_dir
+from easybuild.tools.filetools import copy_file, which, write_file
+from easybuild.tools.modules import get_software_root
+
+
+class EB_BWISE(MakeCp):
+    """
+    Custom easyblock to install BWISE
+    """
+    @staticmethod
+    def extra_options():
+        """Change default values of options"""
+        extra = MakeCp.extra_options()
+        # files_to_copy is not mandatory here
+        extra['files_to_copy'][2] = CUSTOM
+        return extra
+
+    def build_step(self):
+        """Run multiple times for different sources"""
+
+        # BCALM is a git submodule of BWISE but we use it as a dependency
+        bcalm = get_software_root('BCALM')
+        if not bcalm:
+            raise EasyBuildError("BWISE needs BCALM to work")
+
+        makefiles_fixes = [
+            (r'^CC=.*$', 'CC=$(CXX)'),
+            (r'^CFLAGS=.*$', 'CFLAGS:=$(CFLAGS)'),
+            (r'^LDFLAGS=.*$', 'LDFLAGS:=$(LDFLAGS) -fopenmp')
+        ]
+
+        def find_subdir(pattern):
+            """Changes to the sub directory that matches the given pattern"""
+            subdir = glob.glob(os.path.join(self.builddir, pattern))
+            if subdir:
+                return subdir[0]
+            else:
+                raise EasyBuildError("Could not find a subdirectory matching the pattern %s", pattern)
+
+        # BWISE has 3 independant parts, we build them one by one
+        # first BWISE itself
+        subdir = find_subdir(os.path.join('BWISE-*', 'src'))
+        change_dir(subdir)
+        apply_regex_substitutions('makefile', makefiles_fixes)
+        apply_regex_substitutions(os.path.join(subdir, '..', 'Bwise.py'),
+                                  [(r'^BWISE_MAIN = .*$', 'BWISE_MAIN = os.environ[\'EBROOTBWISE\']')])
+        super(EB_BWISE, self).build_step()
```

Comment:
Why not include this in `find_subdir` (you'll need to rename it of course) too, since they are repeated below?

```
change_dir(subdir)
apply_regex_substitutions('makefile', makefiles_fixes)
super(EB_BWISE, self).build_step()
```

You can tweak `Bwise.py` after calling `build_step`?